pkg/didcomm/protocol/introduce/states.go

```diff
@@ -305,27 +314,31 @@ func (s *abandoning) CanTransitionTo(next state) bool {
 	return next.Name() == stateNameDone
 }
 
-func fillMissingDestination(destinations []*service.Destination) []*service.Destination {
-	if len(destinations) <= 1 {
-		return append([]*service.Destination{getInboundDestination()}, destinations...)
+func fillMissingDestination(recipients []*Recipient) []*Recipient {
```

Comment:
should this func be renamed?

Reply:
No, it is fine. The function sets the destination for the inbound message.

Reply:
It seems to be filling in the missing recipient ... that happens to have a destination.

Reply:
so, this function adds recipient(with destination) or sets the destination (if the recipient doesn't have it). But anyway it is related because of the missing destination.